awx/main/management/commands/cleanup_jobs.py

```diff
@@ -57,26 +59,31 @@ def add_arguments(self, parser):
                             action='store_true', dest='only_workflow_jobs',
                             help='Remove workflow jobs')
 
+
     def cleanup_jobs(self):
-        #jobs_qs = Job.objects.exclude(status__in=('pending', 'running'))
-        #jobs_qs = jobs_qs.filter(created__lte=self.cutoff)
         skipped, deleted = 0, 0
-        jobs = Job.objects.filter(created__lt=self.cutoff)
-        for job in jobs.iterator():
-            job_display = '"%s" (%d host summaries, %d events)' % \
-                          (str(job),
-                           job.job_host_summaries.count(), job.job_events.count())
-            if job.status in ('pending', 'waiting', 'running'):
-                action_text = 'would skip' if self.dry_run else 'skipping'
-                self.logger.debug('%s %s job %s', action_text, job.status, job_display)
-                skipped += 1
+
+        batch_size = 1000000
+
+        while True:
+            pk_list = Job.objects.filter(created__lt=self.cutoff).exclude(status__in=['pending', 'waiting', 'running'])[0:batch_size].values_list('pk')
```

Comment:
I think given this that you'll have an infinite loop if you do a dry-run.

Reply:
yep you're right, let me think about a cleaner way to write this

Reply:
fixed